refactor(lang/rust): rust language guide (#310)
bytecodealliance/component-docs: component-model/src/language-support/creating-runnable-components/other-languages.md

# Creating Runnable Components (Other Languages)

This guide is a work in progress and does not have examples for all language toolchains with components support.

For languages not listed in this guide, it is often possible to create runnable components by
following the main principles of the other guides, using the help of the available WebAssembly
toolchain.

Generally, WebAssembly toolchains in the language in question may contain a way to:

> 1. Create a WebAssembly component with the `_start` export (a "command" compnent)

Reviewer: ```suggestion
1. Create a WebAssembly component with the `_start` export (a "command" component)
```

# Creating Runnable Components (Other Languages)

Just because a given programming language is not listed in this guide does not

> mean it is impossible to create runnable WebAssembly components with it.

Reviewer 2: "does not mean it is impossible" feels discouraging to me... not impossible but must be hard if its not here. It feels more inviting to say, "we haven't documented this yet". Suggestion also applies to other `other-language.md` pages.
```suggestion
This guide is a work in progress and does not have examples for all language toolchains with components support.
```

For languages not listed in this guide, it is often possible to create runnable components by
following the main principles of the other guides, using the help of the available WebAssembly
toolchain.

Generally, WebAssembly toolchains in the language in question may contain a way to:

1. Create a WebAssembly component with the `_start` export (a "command" compnent)
2. Create a component that exports the `wasi:cli/run` interface

## Adding a New Language to the Guide

Know of a language guide we should add to this guide? Create a PR to this [repository][repo-pr] that adds 

> the new language guide (similar to others in this section).

Reviewer 2: ```suggestion
## Adding an Example for a Language to this Section

Are you interested in documenting this section for your language toolchain? Create a PR to this [repository][repo-pr] that adds 
the guide (similar to others in this section).
```
